File: examples/c/minilax/minilax.c
# include <stdio.h>
# include "ratc.h"
# include "rMemory.h"
# include "Errors.h"
# include "Scanner.h"
# include "Parser.h"
# include "Tree.h"
# include "Semantic.h"
# include "Types.h"
# include "Defs.h"
# include "ICode.h"
# include "ICodeInt.h"

int main
# ifdef HAVE_ARGS
   (int argc, char * argv [])
# else
   (argc, argv) int argc; char * argv [];
# endif
{
   int ErrorCount;

   StoreMessages (rtrue);
   if (argc > 1) BeginFile (argv [1]);
   ErrorCount = Parser ();

   fprintf (stderr, "Memory%8ld", MemoryUsed);
   fprintf (stderr, "  Tree%8ld", Tree_HeapUsed);
   fprintf (stderr, "  Defs%8ld\n", Defs_HeapUsed);

   BeginTypes ();
   BeginDefs ();
   BeginSemantic ();
   Semantic (TreeRoot);

   fprintf (stderr, "Memory%8ld", MemoryUsed);
   fprintf (stderr, "  Tree%8ld", Tree_HeapUsed);
   fprintf (stderr, "  Defs%8ld\n", Defs_HeapUsed);

   if (ErrorCount == 0) {
      Code (TreeRoot);
/*    WriteCode (); */
      Interpret ();
   }
   WriteMessages (stderr);
   return 0;
}

File: cg/minilax-j/Minilax.java
package minilax;

import de.cocolab.reuse.Global;
import java.io.*;

class Minilax {

    public static void main(String[] argv) throws java.io.IOException {
        int errorCount;
        Scanner scanner = new Scanner ();
        Parser parser = new Parser (scanner);

        Global.errors.storeMessages (true);
        if (argv.length > 0) scanner.beginFile (new FileInputStream (argv [0]));
        errorCount = parser.parse ();

        Semantic.eval (parser.treeRoot);

        if (errorCount == 0) {
            parser.treeRoot.Code ();
            //ICode.WriteCode ();
            ICode.Interpret ();
        }
        Global.errors.writeMessages (System.err);
    }
}
